### PR TITLE
Added CanUndeploy property

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: OpenRA.Mods.Common/Traits/Upgrades/DeployToUpgrade.cs [hunk 1]
```diff
@@ -53,6 +53,9 @@ public class DeployToUpgradeInfo : ITraitInfo, Requires<UpgradeManagerInfo>
 		[Desc("Sound to play when undeploying.")]
 		public readonly string UndeploySound = null;
 
+		[Desc("Is the initial deployment of this actor permanent?")]
+		public readonly bool IsDeploymentPermanent = false;
```
Comment (pchote) [inline OpenRA.Mods.Common/Traits/Upgrades/DeployToUpgrade.cs]: `Permanent` would be sufficient as a name. `Deployment` is redundant, and inconsistent with our usual `Deploy` terminology.

**Comment by dtluna** [comment]
@pchote Will CanUndeploy be better?
P.S. I originally named it that way, but mr. Phrohdoh convinced me to change it to the current variant.

**Comment by pchote** [comment]
CanUndeploy is fine by me.

**Comment by GraionDilach** [comment]
Uhm, what is the point of this? Could you explain the usecase?

**Comment by dtluna** [comment]
@GraionDilach I need the Reckoner APC in my mod to be not able to undeploy.

**Comment by GraionDilach** [comment]
Because? I asked for an explanation, not an example, because all the examples I would think about - notundeployable mobile buildings á lá RA3 Empire of the Rising Sun - are more fitting for `Transforms`. A random unit name from a mod I'm not familiar with ain't explain this to me.

**Comment by dtluna** [comment]
@GraionDilach Because this is how this unit from C&C3:KW works. It is a pretty complex unit and using `Transforms` for this will be not very convinient.

**Comment by GraionDilach** [comment]
Okay, so, now that I looked at the C&C Wikia...

"Because this unit is a mobile bunker which can be deployed to lose mobility in favor of more strength and `Transforms` on this would be messy."

`CanUndeploy` is better, yes.

**Comment by GraionDilach** [comment]
:+1:

**Comment by abcdefg30** [comment]
You probably also want to disable the cursor when you disable undeploying.

**Comment by Mailaender** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/e0039c82813f746304524d0de31708304011f51e...3bd5d502a3b2568874b0e3ab9e0704d601baa872)

**Pull request merged**